### PR TITLE
Auto-repair timed-out agent workflow steps

### PR DESCRIPTION
**Pull request by khaliqgant**
## Summary
- Add deterministic Ricky local auto-fix for Agent Relay agent steps that fail after timeout/retry exhaustion.
- Patch timed-out workflow artifacts by annotating the failed step, inserting a bounded continuation step, and rewiring downstream dependsOn edges to the continuation.
- Keep timeout targeting step-specific first, with run-log fallback only when the timeout failure names the failed step explicitly.
- Preserve complex `timeoutMs` expressions such as `Math.min(MAX_TIMEOUT, 900_000)` when inserting continuation steps.
- Add generation and Workforce persona guidance to split broad implementation/test-writing steps before they become timeout-prone.
- Expose manual resume in the CLI: help shows `ricky run <artifact> --start-from <step>`, and failed run output prints a concrete resume command with `--previous-run-id` when available.
- Add regression coverage for the observed `implement-tests` timeout shape, review-comment edge cases, prompt guardrails, and resume command output.

## Verification
- npm test -- --run src/local/auto-fix-loop.test.ts src/surfaces/cli/commands/cli-main.test.ts src/product/generation/pipeline.test.ts src/product/generation/workforce-persona-writer.test.ts
- npm run typecheck

### CODE REVIEW

File: src/local/auto-fix-loop.ts [hunk 1]
```diff
@@ -484,6 +490,184 @@ function repairUnknownStepTemplateRefs(content: string): { content: string; chan
...
+function timedOutAgentStepFromEvidence(evidence: WorkflowRunEvidence): string | null {
+  const allLogs = evidence.logs.map((entry) => entry.excerpt).join('\n');
+  for (const step of evidence.steps) {
+    if (step.status !== 'failed') continue;
+    const stepText = [
+      step.error,
+      ...step.logs.map((entry) => entry.excerpt),
+      allLogs,
+    ].filter(Boolean).join('\n');
+    if (/\b(?:timed?\s*out|timeout|aborted due to timeout|failed after \d+ retries)\b/i.test(stepText)) {
```
[the remark this answers is not in the document]
Comment (khaliqgant) [reply src/local/auto-fix-loop.ts]: Fixed in 453fcb1. Timeout repair now checks only step-local evidence first and uses run-wide logs only as a fallback when the timeout failure explicitly names the step. Added regression coverage where an earlier non-timeout failed step appears before the timed-out step.

File: src/local/auto-fix-loop.ts [hunk 2]
```diff
@@ -484,6 +490,184 @@ function repairUnknownStepTemplateRefs(content: string): { content: string; chan
...
+  });
+}
+
+function timeoutContinuationPath(content: string, stepId: string): string {
+  if (/\bARTIFACT_DIR\b/.test(content)) return `\${ARTIFACT_DIR}/${stepId}-timeout-continuation.md`;
+  return `.workflow-artifacts/ricky-auto-fix/${stepId}-timeout-continuation.md`;
+}
+
+function timeoutValueForContinuation(block: string): string {
+  const timeout = block.match(/\btimeoutMs:\s*([^,\n}]+)/)?.[1]?.trim();
```
[the remark this answers is not in the document]
Comment (khaliqgant) [reply src/local/auto-fix-loop.ts]: Fixed in 453fcb1. `timeoutMs` is now parsed as a balanced expression, so comma-containing values like `Math.min(MAX_TIMEOUT, 900_000)` are preserved in continuation steps. Added regression coverage for that expression.
